app/controllers/configs_controller.rb

```diff
@@ -2,9 +2,9 @@ class ConfigsController < ApplicationController
   before_action :confirm_admin_user
 
   def index
-    # n+1 join here
-    Config.autosetup
-    @configs = Config.all.sort_by(&:config_key)
+    @configs = Config.config_keys.keys.map do |field|
+      Config.find_or_create_by(config_key: field)
```

Comment:
I'm sorta shocked this is the only spot we do autosetup in!

What do you think about the 'make sure all configs are there on application start' route? I think if we add this to development.rb and production.rb we can get away with it, and do your config map here:

```
  config.after_initialize do
    Config.autosetup
  end
```

Reply:
sounds like a plan! so are those two files the place the do "start-of-app" things?

Reply:
Have this working (will push commit in a few) but realizing one side effect of doing this is that any changes to the configs class require a server reboot - this could make development tricky. we could always turn this ON for prod and OFF for dev.